File: barcodecore-mmo/mythiclib/src/main/java/io/lumine/mythic/lib/api/item/NBTCompound.java
package io.lumine.mythic.lib.api.item;

import java.util.Set;

public abstract class NBTCompound {
    public abstract boolean hasTag(String path);

    public abstract Object get(String path);

    public abstract String getString(String path);

    public abstract boolean getBoolean(String path);

    public abstract double getDouble(String path);

    public abstract int getInteger(String path);

    public abstract NBTCompound getNBTCompound(String path);

    public abstract Set<String> getTags();

    public abstract int getTypeId(String path);
}

File: barcode-framework/platform-bukkit-common/src/main/kotlin/com/vjh0107/barcode/framework/nbt/NBTCompound.kt
package com.vjh0107.barcode.framework.nbt

interface NBTCompound {
    fun hasTag(path: String): Boolean

    operator fun get(path: String): Any?

    fun getString(path: String): String

    fun getBoolean(path: String): Boolean

    fun getDouble(path: String): Double

    fun getInt(path: String): Int

    fun getNBTCompound(path: String): NBTCompound

    fun getTags(): Set<String>

    fun getTypeId(path: String): Int
}